graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java

```diff
@@ -286,12 +291,26 @@ public Response update(@ApiParam(name = "definitionId") @PathParam("definitionId
     public EventDefinitionDto delete(@ApiParam(name = "definitionId") @PathParam("definitionId") @NotBlank String definitionId,
                                      @Context UserContext userContext) {
         checkPermission(RestPermissions.EVENT_DEFINITIONS_DELETE, definitionId);
+
         final Optional<EventDefinitionDto> eventDefinitionDto = dbService.get(definitionId);
+        final String dependencyTitle = eventDefinitionDto.isPresent() ? eventDefinitionDto.get().title() : definitionId;
+        final List<EventDefinitionDto> dependentEventDtoList = eventResolver.dependentEvents(definitionId);
+        if (!dependentEventDtoList.isEmpty()) {
+            final List<String> dependenciesTitles = dependentEventDtoList.stream().map(EventDefinitionDto::title).toList();
+            final List<String> dependenciesIds = dependentEventDtoList.stream().map(EventDefinitionDto::id).toList();
+            String msg = "Unable to delete event definition <" + dependencyTitle + "> because of existing dependencies: " + dependenciesTitles.toString();
```

Comment:
As this is the message that is displayed to the user in the UI, I think we could improve it. What do you think about something like:

```
Unable to delete event definition "A" because it is being referred to by other event definitions. Please remove all references by editing the following event definitions: "B", "C".
```

I'm not sure about the wording, but I think it would be good to give the user a clear indication of what they need to do to fix the issue.

Reply:
If you are doing a bulk delete you can end up with multiple error messages, which get concatenated in the frontend. So it needs to be fairly short. 
Would you prefer this?
`Unable to delete event definition "A" - please remove all references from event definitions: "B", "C"`

Reply:
Ah ok, forgot about bulk actions. I like your suggestion 👍

Reply:
Done